gateway/ln-gateway/src/lib.rs

```diff
@@ -164,6 +171,29 @@ impl LnGateway {
         Ok(())
     }
 
+    async fn assign_channel_id(&self) -> u64 {
+        // The gateway deterministically assigns a channel id (u64) to each federation connected.
+        // At present, the maximum number of federations that can be referenced in the actors map
+        // is equivalent to `usize.max_value()`,
+
+        let mut channels = self.channels.lock().await;
+        let channels_n = channels.len() as u64;
+
+        let actors_n = self.actors.lock().await.len() as u64;
+
+        if actors_n == channels_n {
```

Comment:
When would this not be the case?

Reply:
In the old pattern, `actors_n == channels_n` would only be possible if we did not synchronize updates to the actors hash map and the channels hash set. That would have been a bug in clean up, rather than the expected logic.

Fixed by reworking channel assignment design using [AtomicU64](https://doc.rust-lang.org/std/sync/atomic/struct.AtomicU64.html#) as you suggested. That was a better idea